src/engine/sync/enginesync.cpp

```diff
@@ -308,6 +308,10 @@ void EngineSync::activateMaster(Syncable* pSyncable) {
         qWarning() << "WARNING: Logic Error: Called activateMaster on a NULL Syncable.";
         return;
     }
+    if (m_pMasterSyncable == NULL){
```

Comment:
You're right. I was confused because further below we find `m_pMasterSyncable->getSyncMode()`, but at this point we can be sure that `m_pMasterSyncable` isn't `NULL` since we intercept `pSyncable == NULL` above. Reverting...